torchmetrics/functional/audio/pit.py

```diff
@@ -0,0 +1,209 @@
+    best_indexes = best_indexes.detach()
+    best_perm = ps.T[best_indexes, :]
+    return best_metric, best_perm  # shape [batch], shape [batch, spk]
+
+
+def pit(
+    preds: torch.Tensor,
+    target: torch.Tensor,
+    metric_func: Callable,
+    eval_func: str = 'max',
+    **kwargs: Dict[str, Any]
+) -> Tuple[Tensor, Tensor]:
+    """ Permutation invariant training metric
+
+    Args:
+        target:
+            shape [batch, spk, ...]
+        preds:
+            shape [batch, spk, ...]
+        metric_func:
+            a metric function accept a batch of target and estimate,
+            i.e. metric_func(target[:, i, ...], estimate[:, j, ...]), and returns a batch of metric tensors [batch]
+        eval_func:
+            the function to find the best permutation, can be 'min' or 'max',
+            i.e. the smaller the better or the larger the better.
+        kwargs:
+            additional args for metric_func
+
+    Returns:
+        best_metric of shape [batch],
+        best_perm of shape [batch]
+
+    Example:
+        >>> import torch
+        >>> from torchmetrics.functional.audio import si_sdr, pit, permutate
+        >>> # [batch, spk, time]
+        >>> preds = torch.tensor([[[-0.0579,  0.3560, -0.9604], [-0.1719,  0.3205,  0.2951]]])
+        >>> target = torch.tensor([[[ 1.0958, -0.1648,  0.5228], [-0.4100,  1.1942, -0.5103]]])
+        >>> best_metric, best_perm = pit(preds, target, si_sdr, 'max')
+        >>> best_metric
+        tensor([-5.1091])
+        >>> best_perm
+        tensor([[0, 1]])
+        >>> preds_pmted = permutate(preds, best_perm)
+        >>> preds_pmted
+        tensor([[[-0.0579,  0.3560, -0.9604],
+                 [-0.1719,  0.3205,  0.2951]]])
+
+    Reference:
+        [1]	D. Yu, M. Kolbaek, Z.-H. Tan, J. Jensen, Permutation invariant training of deep models for
+        speaker-independent multi-talker speech separation, in: 2017 IEEE Int. Conf. Acoust. Speech
+        Signal Process. ICASSP, IEEE, New Orleans, LA, 2017: pp. 241–245. https://doi.org/10.1109/ICASSP.2017.7952154.
+    """
+    _check_same_shape(preds, target)
+    if eval_func not in ['max', 'min']:
+        raise RuntimeError('eval_func can only be "max" or "min"')
+    if len(target.shape) < 2:
+        raise RuntimeError(f"Inputs must be of shape [batch, spk, ...], got {target.shape} and {preds.shape} instead")
+
+    # calculate the metric matrix
+    batch_size, spk_num = target.shape[0:2]
+    metric_mtx = torch.empty((batch_size, spk_num, spk_num), dtype=preds.dtype, device=target.device)
+    for t in range(spk_num):
+        for e in range(spk_num):
+            metric_mtx[:, t, e] = metric_func(preds[:, e, ...], target[:, t, ...], **kwargs)
+
+    # find best
+    if spk_num < 3 or _SCIPY_AVAILABLE is None:
+        if spk_num >= 3 and _SCIPY_AVAILABLE is None:
+            import warnings
+            warnings.warn(f"Speaker-num is {spk_num}>3, you'd better install scipy to improve pit's performance!")
```

Comment:
just personal preference 
```suggestion
            warnings.warn(f"In pit metric for speaker-num {spk_num}>3, we recommend installing scipy for better performance")
```